### PR TITLE
Improve error message in VS Setting UI

### PR DESCRIPTION
**Pull request by zhili1208**
Improvement:
1. use NuGetConfigurationException which contains nuget.config path.
2. ask user  to report issue through VSFeedback if unknown issue happens.
3. remove the ConfigInvalid message, because InvalidOperationException is thrown when can't find nuget.config.

### CODE REVIEW

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs [hunk 1]
```diff
@@ -59,14 +59,21 @@ internal void OnActivated()
                     showPackageManagementChooser.Checked = packageManagement.Enabled;
 #endif
                 }
-                catch (InvalidOperationException)
+                // Throwed during creating or saving NuGet.Config.
```
Comment (mishra14) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs]: nit: thrown

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs [hunk 2]
```diff
@@ -59,14 +59,21 @@ internal void OnActivated()
...
+                catch (NuGetConfigurationException ex)
                 {
-                    MessageHelper.ShowErrorMessage(Resources.ShowError_ConfigInvalidOperation, Resources.ErrorDialogBoxTitle);
-                } 
+                    MessageHelper.ShowErrorMessage(ex.Message, Resources.ErrorDialogBoxTitle);
+                }
+                // Throwed if no nuget.config found.
```
Comment (mishra14) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs]: nit: thrown

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs [hunk 3]
```diff
@@ -59,14 +59,21 @@ internal void OnActivated()
...
+                catch (InvalidOperationException ex)
+                {
+                    MessageHelper.ShowErrorMessage(ex.Message, Resources.ErrorDialogBoxTitle);
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs]: Shouldn't this be Resources.ShowError_ConfigInvalidOperation?

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs [hunk 4]
```diff
@@ -59,14 +59,21 @@ internal void OnActivated()
...
+                }
                 catch (UnauthorizedAccessException)
                 {
                     MessageHelper.ShowErrorMessage(Resources.ShowError_ConfigUnauthorizedAccess, Resources.ErrorDialogBoxTitle);
                 }
+                // Unknow exception.
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs]: nit: Unknown

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs [hunk 5]
```diff
@@ -95,16 +102,24 @@ internal bool OnApply()
...
+            catch (NuGetConfigurationException ex)
+            {
+                MessageHelper.ShowErrorMessage(ex.Message, Resources.ErrorDialogBoxTitle);
+                return false;
+            }
+            // Thrown if no nuget.config found.
+            catch (InvalidOperationException ex)
             {
-                MessageHelper.ShowErrorMessage(Resources.ShowError_ConfigInvalidOperation, Resources.ErrorDialogBoxTitle);
+                MessageHelper.ShowErrorMessage(ex.Message, Resources.ErrorDialogBoxTitle);
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs:114]: Why not Resources.ShowError_ConfigInvalidOperation?

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs [hunk 6]
```diff
@@ -95,16 +102,24 @@ internal bool OnApply()
...
                 return false;
             }
             catch (UnauthorizedAccessException)
             {
                 MessageHelper.ShowErrorMessage(Resources.ShowError_ConfigUnauthorizedAccess, Resources.ErrorDialogBoxTitle);
                 return false;
             }
+            // Unknow exception.
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs]: nit: same typo :)

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs [hunk 7]
```diff
@@ -182,14 +183,21 @@ internal void InitializeOnActivated()
...
+            // Thrown during creating or saving NuGet.Config.
+            catch (NuGetConfigurationException ex)
             {
-                MessageHelper.ShowErrorMessage(Resources.ShowError_ConfigInvalidOperation, Resources.ErrorDialogBoxTitle);
+                MessageHelper.ShowErrorMessage(ex.Message, Resources.ErrorDialogBoxTitle);
+            }
+            // Thrown if no nuget.config found.
+            catch (InvalidOperationException ex)
+            {
+                MessageHelper.ShowErrorMessage(ex.Message, Resources.ErrorDialogBoxTitle);
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs:194]: Same  question as the above ones.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs [hunk 8]
```diff
@@ -182,14 +183,21 @@ internal void InitializeOnActivated()
...
             }
             catch (UnauthorizedAccessException)
             {
                 MessageHelper.ShowErrorMessage(Resources.ShowError_ConfigUnauthorizedAccess, Resources.ErrorDialogBoxTitle);
             }
+            // Unknow exception.
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs]: same typo :)

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx [hunk 9]
```diff
@@ -685,7 +682,7 @@ Please see https://aka.ms/troubleshoot_nuget_cache for more help.</value>
     <value>Refresh</value>
   </data>
   <data name="ShowError_ApplySettingFailed" xml:space="preserve">
-    <value>Applying setting failed.</value>
+    <value>Applying setting failed. Please report this issue through VSFeedback.</value>
```
Comment (mishra14) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx]: Failed to apply NuGet Package Manager settings. Please report a problem using Help > Send Feedback > Report a Problem.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs [hunk 10]
```diff
@@ -245,21 +253,24 @@ internal bool ApplyChangedSettings()
...
-                MessageHelper.ShowErrorMessage(ExceptionUtilities.DisplayMessage(e), Resources.ErrorDialogBoxTitle);
+                MessageHelper.ShowErrorMessage(ex.Message, Resources.ErrorDialogBoxTitle);
                 return false;
             }
-            catch (InvalidOperationException)
+            // Thrown if no nuget.config found.
+            catch (InvalidOperationException ex)
             {
-                MessageHelper.ShowErrorMessage(Resources.ShowError_ConfigInvalidOperation, Resources.ErrorDialogBoxTitle);
+                MessageHelper.ShowErrorMessage(ex.Message, Resources.ErrorDialogBoxTitle);
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs:265]: same as above :)

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs [hunk 11]
```diff
@@ -245,21 +253,24 @@ internal bool ApplyChangedSettings()
...
                 return false;
             }
             catch (UnauthorizedAccessException)
             {
                 MessageHelper.ShowErrorMessage(Resources.ShowError_ConfigUnauthorizedAccess, Resources.ErrorDialogBoxTitle);
                 return false;
             }
+            // Unknow exception.
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs]: same typo :)

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx (hunk 9 above)
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx]: > Applying the settings failed

maybe?

Do other call ti VSFeedback or?

**Review by nkolev92 — APPROVED** [review]
Just a few questions and a bunch of nits on typos.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx [hunk 12]
```diff
@@ -800,6 +797,6 @@ Please see https://aka.ms/troubleshoot_nuget_cache for more help.</value>
     <value>Don't show again</value>
   </data>
   <data name="ShowError_SettingActivatedFailed" xml:space="preserve">
-    <value>NuGet setting initialization failed.</value>
+    <value>NuGet Package Manager Settings initialization failed. Please report this issue through VSFeedback.</value>
```
Comment (mishra14) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx]: Failed to initialize NuGet Package Manager Settings. Please report a problem using Help > Send Feedback > Report a Problem.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs (hunk 3 above)
Comment (zhili1208) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs]: we already have the message from Settings

**Review by mishra14 — APPROVED** [review]
Lets improve the messages like we discussed.

**Pull request merged**